src/python/impactx/dashboard/Input/generalFunctions.py

```diff
@@ -52,20 +52,26 @@ def get_default(parameter, type):
     # -----------------------------------------------------------------------------
 
     @staticmethod
-    def determine_input_type(value):
+    def convert_to_numeric(input: str) -> int | float:
         """
-        Determines the type of the input value.
-        :param value: The input value whose type needs to be determined.
-        :return: A tuple containing the value converted to its determined type and the type itself.
+        Converts string inputs to their appropriate numeric type.
+        This method is needed since text fields inputs on the dashboard
+        are inherently strings.
+
+        It first tries to convert the value to int, then to float.
+        If the conversion fails, returns None.
```

Comment:
Could we just add this note, or something along these lines, to the docstring?
```suggestion
        It first tries to convert the value to int, then to float.
        If the conversion fails, returns None.
        Note that the function runs on every keystroke.
        For non-trivial input, e.g., '1e-2', the conversion 
        fails silently until the full number is typed.
```